homeassistant/components/mqtt/tag.py

```diff
@@ -21,10 +23,19 @@
     send_discovery_done,
     update_device,
 )
-from .models import MqttValueTemplate, ReceiveMessage
+from .models import MqttValueTemplate, ReceiveMessage, ReceivePayloadType
 from .subscription import EntitySubscription
 from .util import get_mqtt_data, valid_subscribe_topic
 
+if TYPE_CHECKING:
+
+    class TagComponent:
```

Comment:
I think that this dummy class should be defined as a `Protocol` on the tag integration instead:

```python
if TYPE_CHECKING:
    from homeassistant.components.tag import TagProtocol
```

```python
class TagProtocol(Protocol):
    """Protocol for type checking."""

    async def async_scan_tag(self, tag_id: str, device_id: str, context: Context | None = None) -> None:
        """Handle when a tag is scanned."""
```

Reply:
See #81086

Reply:
Thnx

Reply:
I'll wait until your PR is merged and will then import the TagProtocol for TYPE_CHECKING. BTW MQTT Tag Now allows `None` as `device_id`. I think that should be addressed as well.

Reply:
What is the best way to address?
If the TAGProtocol needs to be adjusted, then please add a comment on my PR.
If the issue is inside MQTT, then it can be resolved after my PR is merged.

Reply:
Moved to https://github.com/home-assistant/core/pull/81495.